iac-common/src/main/java/org/sonar/iac/common/extension/visitors/SyntaxHighlightingVisitor.java

```diff
@@ -40,17 +46,25 @@ protected SyntaxHighlightingVisitor() {
 
   @Override
   protected void before(InputFileContext ctx, Tree root) {
+    inputFile = ctx.inputFile;
     newHighlighting = ctx.sensorContext.newHighlighting()
-      .onFile(ctx.inputFile);
+      .onFile(inputFile);
   }
 
   @Override
   protected void after(InputFileContext ctx, Tree root) {
     newHighlighting.save();
   }
 
-  protected void highlight(HasTextRange range, TypeOfText typeOfText) {
-    newHighlighting.highlight(range.textRange(), typeOfText);
+  protected void highlight(HasTextRange tree, TypeOfText typeOfText) {
+    if (tree.textRange().end().compareTo(tree.textRange().start()) <= 0) {
```

Comment:
The check was added after seeing noticing [YamlHighlightingVisitorTest#scalar_key](https://github.com/SonarSource/sonar-iac/blob/32ba45207106a08be9fe3b65a05283c701ae2a39/iac-common/src/test/java/org/sonar/iac/common/yaml/visitors/YamlHighlightingVisitorTest.java#L50) failing. Reason is that [InputFile#newRange](https://github.com/SonarSource/sonar-plugin-api/blob/af6f59056ad3258572ca76c47f4802b54063bfa2/plugin-api/src/main/java/org/sonar/api/batch/fs/InputFile.java#L181) actually checks that the range makes sense and raises an exception if not. 
In Yaml we currently parse `key:` to a `Mapping` with a `Tuple` whose key is a `ScalarTree` `"key:"` and it's value is a `ScalarTree` that represents an empty String and hence startColumn == endColumn